components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppInterceptor.kt

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.pwa
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import kotlinx.coroutines.runBlocking
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.request.RequestInterceptor
+import mozilla.components.feature.intent.ext.putUrlOverride
+import mozilla.components.feature.pwa.intent.WebAppIntentProcessor
+
+/**
+ * This feature implements use cases for detecting and handling redirects to standalone web app activities.
+ *
+ * In the case of Android Intents that are not installed, and with no fallback, the user is prompted
+ * to search the installed market place.
+ *
+ * @param shortcutManager current shortcut manager instance to lookup web app install states
+ */
+@Suppress("LongParameterList")
```

Comment:
Move the suppress call to the `onLoadRequest` function instead of the entire class